var/spack/repos/builtin/packages/assimp/package.py

```diff
@@ -37,5 +37,15 @@ class Assimp(CMakePackage):
     depends_on('boost')
 
     def cmake_args(self):
-        args = ['-DASSIMP_BUILD_TESTS=OFF']
+        args = [
+            '-DASSIMP_BUILD_TESTS=OFF',
+            '-DBUILD_SHARED_LIBS:BOOL=%s' % (
+                'ON' if '+shared' in self.spec else 'OFF'),
```

Comment:
hm, but it does not have `shared` variant, so this is always `OFF`, right?

Reply:
For some reason it made the difference.

Also - I copied this from var/spack/repos/builtin/packages/amp/package.py - which also does not have a `shared` variant

Reply:
i see that it does make a difference, I just expect this to result in `-DBUILD_SHARED_LIBS:BOOL=OFF`. Out of curiosity, could you please check the logs?

Reply:
Yes - I see now that its using `-DBUILD_SHARED_LIBS:BOOL=OFF` after this change

```
==> Executing phase: 'cmake'
==> 'cmake' '/home/balay/spack.x/var/spack/stage/assimp-4.0.1-jkepmgkwhvnjycepaylo3rnte7px22qm/assimp-4.0.1' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:PATH=/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/assimp-4.0.1-jkepmgkwhvnjycepaylo3rnte7px22qm' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH:STRING=/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/assimp-4.0.1-jkepmgkwhvnjycepaylo3rnte7px22qm/lib;/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/assimp-4.0.1-jkepmgkwhvnjycepaylo3rnte7px22qm/lib64;/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/boost-1.68.0-75pmeimn6vug4em6pmvu7fdoi6a3d2ji/lib;/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/bzip2-1.0.6-efym4s2lzs2mnm7ecujmqnnyn2hotvpk/lib;/usr/lib;/usr/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/cmake-3.12.1-bf6wtchrie5warssfifbqjormyw2mrv6;/home/balay/spack.x/opt/spack/linux-centos7-x86_64/intel-18.0.2/boost-1.68.0-75pmeimn6vug4em6pmvu7fdoi6a3d2ji' '-DCMAKE_CXX_FLAGS=-xMIC-AVX512 -O3 -fp-model fast -std=c++11' '-DCMAKE_Fortran_FLAGS=-xMIC-AVX512 -O3 -fp-model fast' '-DCMAKE_C_FLAGS=-xMIC-AVX512 -O3 -fp-model fast' '-DASSIMP_BUILD_TESTS=OFF' '-DBUILD_SHARED_LIBS:BOOL=OFF'
-- The C compiler identification is Intel 18.0.2.20180210
-- The CXX compiler identification is Intel 18.0.2.20180210
```

Reply:
On OSX build I now get deal build errors - presumably due to assimp now being static
[dealii-spack-build.out.txt](https://github.com/spack/spack/files/2362644/dealii-spack-build.out.txt)

Reply:
Probably because if it’s static, I should be build with fPIC, could you add this please for static builds?

Reply:
yes, so  I pushed a fix making 'shared' default for assimp in this PR

Reply:
Somehow my my KNL test is broken - so I'm restarting it. It takes a very long time to complete the xsdk build including dealii.  Will merge this PR - after its done.

Reply:
I can confirm the build is with: '-DBUILD_SHARED_LIBS:BOOL=ON'

but then - I get:

```
ld: ../contrib/irrXML/libIrrXML.a(irrXML.cpp.o): relocation R_X86_64_32 against `.rodata.str1.4' can not be used when making a shared object; recompile with -fPIC
ld: final link failed: Nonrepresentable section on output
make[2]: *** [code/libassimp.so.4.0.1] Error 1
```

[assimp-spack-build.out.txt](https://github.com/spack/spack/files/2363736/assimp-spack-build.out.txt)

So its likely a bug within assimp build tools.

And this patch doesn't help xsdk [but it improves assimp build slightly - so we should merge this?]

I'll need  #9190 anyway

Reply:
> so we should merge this?

i think so, it still improves the package.

Reply:
ok merging now.